Index: stterm/config.def.h
===================================================================
--- stterm.orig/config.def.h	2019-04-13 22:17:56.035224839 +0100
+++ stterm/config.def.h	2019-04-13 22:18:21.567452877 +0100
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "Hack:pixelsize=16:antialias=true:autohint=true";
 static int borderpx = 2;
 
 /*
